### PR TITLE
Bluetooth: audio: ascs: Fixes and improvements for service and tests

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This fixes potential null pointer dereference in bap_stream and unexpected assert in ASCS.
The ASCS unit tests have been improved by additional tests and minor fixes.

### CODE REVIEW

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -1793,19 +1785,32 @@ static int ase_stream_qos(struct bt_bap_stream *stream, struct bt_audio_codec_qo
 		LOG_WRN("Invalid operation in state: %s", bt_bap_ep_state_str(ep->status.state));
 		*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_INVALID_ASE_STATE,
 				       BT_BAP_ASCS_REASON_NONE);
-		return -EBADMSG;
+		return;
+	}
+
+	stream = ep->stream;
+	CHECKIF(stream == NULL) {
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:1792]: `CHECKIF` should IMO only be used to check for parameters in non-static functions. 

```suggestion
	if (stream == NULL) {
```

File: subsys/bluetooth/audio/ascs.c [hunk 2]
```diff
@@ -1793,19 +1785,32 @@ static int ase_stream_qos(struct bt_bap_stream *stream, struct bt_audio_codec_qo
...
+		LOG_ERR("NULL stream");
+		*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_UNSPECIFIED, BT_BAP_ASCS_REASON_NONE);
+		return;
+	}
+
+	CHECKIF(stream->ep == NULL) {
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:1798]:
```suggestion
	if (stream->ep == NULL) {
```

File: tests/bluetooth/audio/ascs/prj.conf [hunk 3]
```diff
@@ -5,11 +5,10 @@ CONFIG_BT_MAX_CONN=1
 CONFIG_BT_ISO_MAX_CHAN=1
 CONFIG_BT_AUDIO=y
 CONFIG_BT_ASCS=y
-CONFIG_BT_ASCS_ASE_SNK_COUNT=2
-CONFIG_BT_ASCS_ASE_SRC_COUNT=2
+CONFIG_BT_ASCS_ASE_SNK_COUNT=1
+CONFIG_BT_ASCS_ASE_SRC_COUNT=1
```
Comment (Thalley) [inline tests/bluetooth/audio/ascs/prj.conf:9]: We don't have any tests that rely on multiple streams?

**Review by Thalley — CHANGES_REQUESTED** [review]

File: tests/bluetooth/audio/ascs/prj.conf (hunk 3 above)
Comment (MariuszSkamra) [reply tests/bluetooth/audio/ascs/prj.conf:9]: Nope

Comment (Thalley) [reply tests/bluetooth/audio/ascs/prj.conf:9]: We should consider that at some point - I guess none of the for loops to handle multiple ASEs in a single GATT write command is properly exercised currently

**Review by Thalley — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**